### ISSUE TITLE
Upcoming change: Wifi firmware removal

### ISSUE THREAD

**Issue by travier** [ublue-os/ucore#138]
As part of https://github.com/coreos/fedora-coreos-tracker/issues/1575, we are planning to remove the remaining wireless card firmwares from the Fedora CoreOS image.

To make sure to keep all those firmware in the ucore images, you will likely want to pre-emptively list them here.

You might also want to include `NetworkManager-wifi` for full wifi support.

And update the doc: https://github.com/ublue-os/ucore#ucore-1

For the full list, see: https://github.com/coreos/fedora-coreos-config/pull/2938

See also: https://github.com/coreos/fedora-coreos-tracker/issues/862

**Comment by travier** [comment]
You might also want to add `iwd` (https://iwd.wiki.kernel.org/, https://wiki.archlinux.org/title/iwd) which is more stable than `wpa_supplicant` in my experience.

**Comment by bsherman** [comment]
@travier thank you for the issue and the PRs! I appreciate the support from you and the CoreOS team!

I saw you updated #139 so let me know when you think that list is stable and I can proceed with merging.

It seems the outcome of this will be a responsibility for ucore to track changes of the wifi firmware names in Fedora, but that seems reasonable.

Regarding `NetworkManager-wifi` and `iwd` ... I have not used on `iwd` on Fedora yet, but thanks for the tip. I'll do some testing with that, but we may as well add it with `NetworkManager-wifi` when the time comes.

**Comment by bsherman** [comment]
Small update on `iwd` testing... I've been using on my custom Silverblue images the past couple days and while it has been positive, I wasn't having issues before either, so I'm not sure what to think. 

One thing I did notice, I had to forget/rejoin NetworkManager saved wifi networks.  I first tried modifying the the `foo.connection` files to remove the specified wifi device since that changed from predictable naming (eg, `wlp8s0`) to `wlan0`, of course restarting NetworkManager after doing so. But this wasn't enough, the network would not join.  I still had to forget/join the network again. I think this is likely due to `iwd` storing credentials in `/var/lib/iwd` even when used as a backend for NetworkManager-wifi.

TL;DR
If `iwd` is truly better, I wish Fedora upstream would default to it.
As is, I'd be hesitant to change to it for ublue desktops given users would probably have a frustrating transition without very clearly setting expectations.
If we want to use `iwd` by default in `ucore` we should probably do so from day one, to avoid the frustrating transition problem in ucore.

**Comment by travier** [comment]
> I saw you updated #139 so let me know when you think that list is stable and I can proceed with merging.

This list should be accurate right now.

> It seems the outcome of this will be a responsibility for ucore to track changes of the wifi firmware names in Fedora, but that seems reasonable.

Yes, this is an unfortunate consequence of the firmware package split. Hopefully this does not happen too often.

> TL;DR If `iwd` is truly better, I wish Fedora upstream would default to it. As is, I'd be hesitant to change to it for ublue desktops given users would probably have a frustrating transition without very clearly setting expectations. If we want to use `iwd` by default in `ucore` we should probably do so from day one, to avoid the frustrating transition problem in ucore.

Agree that doing the work upstream is better.
